### PR TITLE
Internal: Github-Workflow: Run lint and tests only if necessary [ED-3255]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -29,21 +29,30 @@ jobs:
     steps:
       - name: Checkout source code
         uses: actions/checkout@v2
+      - name: Check PHP files diff
+        uses: technote-space/get-diff-action@v4
+        with:
+          PATTERNS: |
+            **/*.php
       - name: Setup PHP 7.4
+        if: ${{ env.GIT_DIFF }}
```
Comment (matipojo) [inline .github/workflows/lint.yml:38]: Try to find or create an action that sets the test as passed and stops it?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -8,7 +8,13 @@ jobs:
     steps:
       - name: Checkout source code
         uses: actions/checkout@v2
+      - name: Check JS files diff
+        uses: technote-space/get-diff-action@v4
+        with:
+          PATTERNS: |
+            **/*.(js|ts|json)
```
Comment (matipojo) [inline .github/workflows/lint.yml:15]: jsx

**Review by matipojo — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (Nevoss) [reply .github/workflows/lint.yml]: Done

**Review by matipojo — APPROVED** [review]

**Review by ronkelementor — DISMISSED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Fix conflict

**Review by matipojo — CHANGES_REQUESTED** [review]
Fix conflict

File: .github/workflows/js-qunit.yml [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+        uses: actions/checkout@v2
+      - name: Check JS files diff
+        id: js_diff_files
+        uses: technote-space/get-diff-action@v4
+        with:
+          PATTERNS: |
+            **/*.+(js|ts|json|jsx|tsx)
+
+  q-unit:
+    runs-on: ubuntu-16.04
```
Comment (matipojo) [inline .github/workflows/js-qunit.yml:23]: Here and next
```suggestion
    runs-on: ubuntu-18
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: .github/workflows/js-qunit.yml (hunk 3 above)
Comment (Nevoss) [reply .github/workflows/js-qunit.yml:23]: Done

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]

**Pull request merged**